src/api.calls.ts

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios'
+
+type SpriteType = string | null
+
+export type Pokemon = {
+  name: string
+  id: number
+  sprites: {
+    front_default: SpriteType
+    front_shiny: SpriteType
+    front_female: SpriteType
+    front_shiny_female: SpriteType
+    back_default: SpriteType
+    back_shiny: SpriteType
+    back_female: SpriteType
+    back_shiny_female: SpriteType
+  }
+}
+
+type PokemonType = {
+  slot: number
+  type: {
+    name: string
+  }
+}
+
+type PokemonAbility = {
+  ability: { name: string }
+  slot: number
+}
+
+export type EnhancedPokemon = Pokemon & { abilities: PokemonAbility[]; pokemonTypes: PokemonType[] }
+
+type PokemonBatchResponse = {
+  name: string
+  url: string
+}
+const api = 'https://pokeapi.co/api/v2/pokemon/'
+
+// search for a get a single pokemon
+export async function getPokemon(searchQuery: string): Promise<Pokemon[]> {
+  try {
+    const res = await axios.get(api + searchQuery)
+    const pokemon = {
```

Comment:
You could do:

const { name, id, sprites, abilities, perkemonTypes } = res
return [{ name, id, sprites, abilities, perkemonTypes }]

so you don't have to repeat res.data.